google/auth/_helpers.py

```diff
@@ -88,3 +89,46 @@ def from_bytes(value):
     else:
         raise ValueError(
             '{0!r} could not be converted to unicode'.format(value))
+
+
+def update_query(url, params):
+    """Updates a URL's query parameters.
+
+    Replaces any current values if they are already present in the URL.
+
+    Args:
+        url (str): The URL to update.
+        params (Mapping[str, Optional[str]]): A mapping of query parameter
+            keys to values. If the value of a parameter is `None`, the query
+            parameter will be removed from the query string.
+
+    Returns:
+        str: The URL with updated query parameters.
+
+    Examples:
+
+        >>> url = 'http://example.com?a=1'
+        >>> update_query(url, {'a': '2'})
+        http://example.com?a=2
+        >>> update_query(url, {'b': '3'})
+        http://example.com?a=1&b=3
+        >> update_query(url, {'a': None, 'b': '3'})
```

Comment:
I don't like it, but since you seem really opposed to the current implementation I'll switch it.